src/internal/packager2/layout/package.go

```diff
@@ -121,8 +121,14 @@ func (p *PackageLayout) Cleanup() error {
 	return nil
 }
 
+// ErrNoSBOMAvailable is returned when a user tries to access a package SBOM, but it is not available
+var ErrNoSBOMAvailable = errors.New("zarf package does not have an SBOM available")
+
 // GetSBOM outputs the SBOM data from the package to the give destination path.
 func (p *PackageLayout) GetSBOM(destPath string) (string, error) {
+	if !p.Pkg.IsSBOMAble() {
+		return "", ErrNoSBOMAvailable
+	}
```

Comment:
Let's declare `ErrNoSBOMAvailable` as a custom error type rather than a named error string.  The message could pass back the package name or sha as metadata e.g. :

```
...
(e *NoSBOMAvailableError) Error() string {
    return fmt.Sprintf("zarf package does not have an SBOM available, package=%s", e.package)
}
```
https://gobyexample.com/custom-errors

Reply:
I like this, will implement now